### PR TITLE
feat: live update `/queue`

### LINKED ISSUE #92: Live update queue command
Changes to the queue (either new tracks being added or a track that finished playing) should automatically update the queue response interaction instead of the user having to click the arrow buttons to refresh it.

Visually, the queue button components could use a little change:
* Disable buttons when it's not possible to navigate to a specific page
* Use alt-codes instead of emojis
* Use primary blurple color instead of grey

## Current
image

## Expected
image

### PR DESCRIPTION
**Pull request by aquelemiguel**
Aims to close #92 and #94.

### CODE REVIEW

File: src/client.rs
Comment (joao-conde) [inline src/client.rs:19]: Move to its own module. In fact, wait for the approval and merge of https://github.com/aquelemiguel/parrot/pull/98 and let's merge this with the guild settings we have (possibly even rename to `GuildCache`)

File: src/commands/clear.rs
Comment (joao-conde) [inline src/commands/clear.rs:38]: Just an idea, maybe send the "Clear!" message first and then update the queue

File: src/commands/play.rs
Comment (joao-conde) [inline src/commands/play.rs:68]: Again, I would update the queue at the end of this command, I think feedback to the command is important first.

File: src/commands/play.rs
Comment (joao-conde) [inline src/commands/play.rs:126]: This is not true for `/playtop` right?

File: src/commands/queue.rs
Comment (joao-conde) [inline src/commands/queue.rs:31]: When this reaches 0 and it timeouts, does the embed disappear, like do we remove the embed message? If so, I would prefer `EMBED_TIMEOUT` and maybe with a little docstring specifying the time units (I know its seconds but still)

File: src/commands/queue.rs
Comment (joao-conde) [inline src/commands/queue.rs:68]: In reality, this is the `page`. Only when you call `page.read()` or `page.write()` do you get a lock. In the former a `page_rlock` in the latter a `page_wlock`

File: src/commands/queue.rs
Comment (joao-conde) [inline src/commands/queue.rs:105]: And this would be

```rust
let mut page_wlock = page.write().await;
```

File: src/commands/queue.rs
Comment (joao-conde) [inline src/commands/queue.rs:173]: Is this not possible?

```suggestion
    CreateButton::default()
        .custom_id(label.to_string().to_ascii_lowercase())
        .label(label)
        .style(ButtonStyle::Primary)
        .disabled(is_disabled)
```

File: src/commands/queue.rs
Comment (joao-conde) [inline src/commands/queue.rs:183]: Nice. Except for the triggering NEW LINE *triggered*

File: src/commands/remove.rs
Comment (joao-conde) [inline src/commands/remove.rs:55]: Again, maybe first send the feedback to the user ("Removed!") and then update the queue embed

File: src/commands/shuffle.rs
Comment (joao-conde) [inline src/commands/shuffle.rs:31]: Same idea, feedback to the user first.

File: src/commands/stop.rs
Comment (joao-conde) [inline src/commands/stop.rs:35]: Feedback first

File: src/events/modify_queue_handler.rs
Comment (joao-conde) [inline src/events/modify_queue_handler.rs:28]: In reality, you only use this handler for the `Event::Track(TrackEvent::End)` so you know the `EventContext` will be of type Track. Since you don't use anything, I think you can remove the `if let`

File: src/events/modify_queue_handler.rs
Comment (joao-conde) [inline src/events/modify_queue_handler.rs:60]: This is all to compute size shrunk, just put it together, why the new line

File: src/tests/utils.rs
Comment (joao-conde) [inline src/tests/utils.rs:1]: Weird that clippy does not complain, but I think you do not need this import

File: src/tests/utils.rs
Comment (joao-conde) [inline src/tests/utils.rs:7]: Add a test case where it receives a `None`

File: src/utils.rs
Comment (joao-conde) [inline src/utils.rs:45]: This can better be rewritten as a match now that we have three cases: Some duration and infinite, some duration and finite, and the rest

**Review by joao-conde — CHANGES_REQUESTED** [review]

File: src/commands/play.rs
Comment (aquelemiguel) [reply src/commands/play.rs:126]: Yeah, you're right. Oversight by my part, it'll show "Estimated time until play: ∞" which is incorrect.

File: src/commands/queue.rs
Comment (aquelemiguel) [reply src/commands/queue.rs:173]: Tried it as well, it returns a `&mut CreateButton`. Could slap it a `.to_owned` but I dunno.

File: src/commands/queue.rs
Comment (aquelemiguel) [reply src/commands/queue.rs:183]: Looks a lot worse on GitHub than VSCode, lol.

File: src/commands/queue.rs
Comment (aquelemiguel) [reply src/commands/queue.rs:31]: The embed gets edited with `QUEUE_EXPIRED`, seen here:

https://github.com/aquelemiguel/parrot/blob/cd54d3432bf957022736a58007cf4bf20a9ea0d3/src/commands/queue.rs#L125-L135

Comment (joao-conde) [reply src/commands/queue.rs:31]: Then I would say just to use `EMBED_TIMEOUT`

File: src/commands/queue.rs
Comment (joao-conde) [reply src/commands/queue.rs:173]: I like that

**Comment by joao-conde** [comment]
Haven't forgotten but will wait for merge conflicts fix 👍

File: src/commands/shuffle.rs
Comment (afonsojramos) [inline src/commands/shuffle.rs:32]: Can you move the text to `strings.rs`?

File: src/commands/stop.rs
Comment (afonsojramos) [inline src/commands/stop.rs:34]: Can you move the text to `strings.rs`?

File: src/commands/clear.rs
Comment (afonsojramos) [inline src/commands/clear.rs:37]: Can you move the text to `strings.rs`?

**Review by afonsojramos — CHANGES_REQUESTED** [review]

File: src/utils.rs
Comment (afonsojramos) [inline src/utils.rs:83]: This function should be deleted as it is now in `play.rs`.

**Review by afonsojramos — CHANGES_REQUESTED** [review]

**Review by joao-conde — APPROVED** [review]

**Review by afonsojramos — APPROVED** [review]

**Pull request merged**
